languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/HttpRequest.java

```diff
@@ -0,0 +1,52 @@
+package client;
+
+public class HttpRequest {
```

Comment:
i would suggest removing the setters/getters here since they would be an annoyance for type completion, my idea is that all is set at constructor in an immutable fashion.

Reply:
since we will add for example .withExpansion() to requests, how would we then change the path if everything is set in the constructor on creation time? e.g : Api.products().get().withExpansion(). Inside of withExpansion I have to call setPath. Also I might want to change headers. e.g : Request r = Api.products().get(); r.setHeader("X-Correlation-ID", generateId()). We can use addHeader() instead of setHeader so that it stays a builder pattern but we still need a setHeaders() method.

Reply:
The way i did it is that expansions, headers.... are all arguments to the method 
i think its elegant and nice solution.